honeybadgermpc/ipc.py

```diff
@@ -222,26 +222,32 @@ def get_send_and_recv(self, sid):
 
         def make_send(i, sid):
             def _send(j, o):
-                logging.debug('[%s] SEND %8s [%2d -> %2d]' % (sid, o[1], i, j))
+                logging.debug('[%s] SEND %8s [%2d -> %2d]' % (sid, o, i, j))
                 if i == j:
                     # If attempting to send the message to yourself
                     # then skip the network stack.
                     listener_queue.put_nowait((i, o))
                 else:
                     self.senders.queues[j].put_nowait((sid, (i, o)))
-            return _send
+
+            def _broadcast(o):
+                for r in range(self.N):
+                    _send(r, o)
+
+            return _send, _broadcast
 
         def make_recv(j, sid):
             async def _recv():
                 (i, o) = await self.listener.get_message(sid)
-                logging.debug('[%s] RECV %8s [%2d -> %2d]' % (sid, o[1], i, j))
+                logging.debug('[%s] RECV %8s [%2d -> %2d]' % (sid, o, i, j))
                 return (i, o)
             return _recv
 
-        return make_send(self.nodeid, sid), make_recv(self.nodeid, sid)
+        send, broadcast = make_send(self.nodeid, sid)
+        return send, make_recv(self.nodeid, sid), broadcast
 
     def add(self, sid, program, **kwargs):
-        send, recv = self.get_send_and_recv(sid)
+        send, recv, _ = self.get_send_and_recv(sid)
```

Comment:
I don't think we should modify `get_send_and_recv` to return `broadcast`, that can just be locally defined by whoever wants to use it. It's really better called "multicast" anyway.

Reply:
Changed.